src/engraving/rendering/dev/paint.cpp

```diff
@@ -129,21 +131,14 @@ void Paint::paintScore(draw::Painter* painter, Score* score, const IScoreRendere
             }
 
             // Draw page elements
-            bool disableClipping = false;
-
-            if (!painter->hasClipping()) {
-                painter->setClipping(true);
-                painter->setClipRect(pageRect);
-                disableClipping = true;
```

Comment:
It's also causing this regression:

<img width="635" alt="Screen Shot 2024-04-08 at 10 18 13" src="https://github.com/musescore/MuseScore/assets/67867444/53c20fc7-61eb-496d-9d5d-3e3ae0ac2a1e">

Reply:
I think I've also seen that in master sometimes

Reply:
How do you reproduce that?  I couldn't, using a build from my own branch.

Reply:
Hmm ok I *can* reproduce it (it only seems to happen at particular zoom levels), but it *does* also happen using the version from master - just not quite as obviously, and only if you zoom in and out while playing back. Will investigate.

Reply:
Wow, this has me totally baffled, nothing makes sense as to when and why this happens and what to do about it (for me most of the time zooming into 141% then starting playback is enough, but sometimes it doesn't happen at that level and you need to zoom in even further).  It's hard not to suspect there's a bug in Qt somewhere, but I agree it's definitely more likely to happen with the changes in this PR than on master, and I'd even accept it's likely to affect more users than the bug that it was trying to fix (seriously as that is).

Reply:
Ok, found something that works, have a slightly better understanding now of why the existing clipping rectangle is getting set (though far from perfect), but it seems that the safest/most reliable/performant fix is to simply treat the case of the drawing rectangle extending below the page as a special case and in this case overriding the clipping rectangle with the page rectangle (intersecting was causing other issues).